### PR TITLE
Fix focus scope restore losing focus

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Go to DialogTrigger -> with Menu Trigger, use the keyboard to open the Menu, press tab, focus should be restore to the Menu Trigger button

### CODE REVIEW

**Review by ktabors — APPROVED** [review]
Like that you wrote a test that mirrors the reported bug.
Tested dialog and focus scope on Firefox and Chrome.
LGTM

File: packages/@react-aria/focus/src/FocusScope.tsx [hunk 1]
```diff
@@ -376,8 +376,16 @@ function useRestoreFocus(scopeRef: RefObject<HTMLElement[]>, restoreFocus: boole
         if (nextElement) {
           focusElement(nextElement, true);
         } else {
-          // If there is no next element, blur the focused element to move focus to the body.
-          focusedElement.blur();
+          if (contain) {
```
Comment (devongovett) [inline packages/@react-aria/focus/src/FocusScope.tsx:379]: Isn't this backwards? If it's contained, then we _don't_ want focus to go back to the body, right?

File: packages/@react-aria/focus/src/FocusScope.tsx [hunk 2]
```diff
@@ -376,8 +376,16 @@ function useRestoreFocus(scopeRef: RefObject<HTMLElement[]>, restoreFocus: boole
...
+            // If there is no next element, blur the focused element to move focus to the body.
+            focusedElement.blur();
+          } else {
+            requestAnimationFrame(() => {
```
Comment (devongovett) [inline packages/@react-aria/focus/src/FocusScope.tsx:383]: Is the raf necessary here?

**Comment by snowystinger** [comment]
Thanks @LFDanLu you can approve for me since I can't approve it
I tested in Chrome/Safari/FF

**Review by LFDanLu — APPROVED** [review]
proxy approval 😄

**Review by devongovett — APPROVED** [review]

**Pull request merged**